Applications/Spire/Include/Spire/Login/LoginWindow.hpp

```diff
@@ -82,7 +82,8 @@ namespace Spire {
       QLineEdit* m_username_line_edit;
       QLineEdit* m_password_line_edit;
       FlatButton* m_sign_in_button;
-      IconButton* m_exit_button;
+      Button* m_close_button;
+      boost::signals2::scoped_connection m_close_button_connection;
```

Comment:
Why is the connection needed here?

Reply:
The scoped_connection is disconnected when it's destructed, but the connection returned by the connect_x_signal functions isn't disconnected when destructed. This can create problems when object that is connected to the signal is destroyed, the slot connected to that signal may no longer exist.

In this case it probably won't make a difference because the TextBox will be destroyed before the LoginWindow so there's no chance for the the slot to be null, but it's an informal standard to store the returned connection as a scoped_connection to prevent issues like this.

Reply:
In our project, some returned connections of the connect_x_signal function are stored, but others are not saved. If the issue that you said can be avoided, is it necessary to store it? Is there a guide for this?

Reply:
Good question. If the signaller outlives the receiver, then the receiver must keep a ```scoped_connection``` to the signaller to avoid a dangling reference. A common example is a widget that connects to a model; models are shared among many widgets so it's very likely that a model outlives a widget and hence widgets often have ```scoped_connection```s to models. 

In this case, however, the ```LoginWindow``` outlives the ```m_close_button``` so no ```scoped_connection``` is needed.

Reply:
I always store the connection, so maybe @kranar can expand on cases where we wouldn't store them.

The [boost documentation](https://www.boost.org/doc/libs/1_63_0/doc/html/signals/tutorial.html#idp607875008) has some info on connections, but not specifically on best practices.

If we passed a function pointer to the connect_x_signal functions, we might not have to store the connections because they would be disconnected when the object connecting to the signal is destroyed. I think our use of lambdas as slots prevents this automatic disconnection because the signal can't determine the state of the objects in the lambda slot.

Reply:
I see. Thanks @kranar @spireken

Reply:
@kranar Should we say then, that in general for signals from a child widget to a parent widget slot don't bother storing the connection?

Reply:
Yeah that works for most cases.